modules/client-common/src/internal/modules/web3.ts

```diff
@@ -138,11 +137,124 @@ export class Web3Module implements IClientWeb3Core {
     });
   }
 
+  /** FRAMEWORK ADDRESSES */
   /** Returns the current DAO factory address */
-  public getDaoFactoryAddress(): string {
-    if (!this.context.daoFactoryAddress) {
-      throw new NoDaoFactory();
+  get daoFactoryAddress(): string {
+    const daoFactoryAddress = this.context.daoFactoryAddress;
+    if (!daoFactoryAddress || !isAddress(daoFactoryAddress)) {
+      throw new InvalidAddressError();
     }
-    return this.context.daoFactoryAddress;
+    return daoFactoryAddress;
+  }
+
+  /** Returns the current plugin setup processor address */
```

Comment:
```suggestion
  frameworkAddress(addressName: string): string {
    
    const frameworkddress = this.context[addressName];
    if (!frameworkddress || !isAddress(frameworkddress)) {
      throw new InvalidAddressError();
    }
    return frameworkddress;
  }
```

addressName should be its own type